File: backend/app/core/upgrades/stat_modifiers/player_health_5.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerHealth5(UpgradeImplementation):
    id = 'player_health_5'
    name = 'Reinforced Core'
    description = '+300 max health'
    rarity = 'legendary'
    type = 'stat_modifier'
    cost = 40

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 300
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/player_health_5.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerHealth5Def: UpgradeDef = {
  id: "player_health_5",
  name: "Reinforced Core",
  description: "+300 max health",
  rarity: RarityID.Legendary,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 40,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: 300,
  stackable: true,
  maxStacks: 99999999,
}

export class PlayerHealth5 extends Upgrade {}
